src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -468,17 +484,45 @@ private BoundExpression MakeConversionNode(BoundExpression rewrittenOperand, Typ
                      rewrittenOperand.Type.SpecialType != SpecialType.System_DateTime)
                 {
                     // error CS0029: Cannot implicitly convert type '{0}' to '{1}'
-                    _diagnostics.Add(
+                    diagnostics.Add(
                         ErrorCode.ERR_NoImplicitConv,
                         rewrittenOperand.Syntax.Location,
                         rewrittenOperand.Type,
                         rewrittenType);
                 }
+            }
 
-                return _factory.NullOrDefault(rewrittenType);
+            return conversion;
+        } 
+
+        private static BoundExpression MakeConversionNodeIfNecessaryWithoutLowering(
+            BoundExpression operand, 
+            TypeSymbol type, 
+            SyntaxNode syntax, 
+            CSharpCompilation compilation, 
+            DiagnosticBag diagnostics, 
+            bool @checked, 
+            bool acceptFailingConversion = false)
+        {
+            if (operand.Type.Equals(type))
```

Comment:
>if (operand.Type.Equals(type)) [](start = 12, length = 30)

I think this check should be based on whether ```conversion``` (calculated below) is identity conversion or not.